src/eventreceiver.ts

```diff
@@ -0,0 +1,47 @@
+import { Tracer } from './telemetry/traces';
+import { GlobalLogger as Logger } from './telemetry/logs';
+import { WorkflowFunction, WorkflowHandle, WorkflowParams } from './workflow';
+import { Transaction } from './transaction';
+import { MethodRegistrationBase } from './decorators';
+
+/*
+ * Info provided to an event receiver at initialization,
+ *  which contains the things that it needs to do its work
+ *  (retrieve decorated endpoints, and run new transactions / workflows)
+ */
+export interface DBOSExecutorEventReceiverInterface
+{
+  /* Logging service */
+  readonly logger: Logger;
+  /* Tracing service */
+  readonly tracer: Tracer;
+
+  getConfig<T>(key: string): T | undefined;
+  getConfig<T>(key: string, defaultValue: T): T;
+
+  /*
+   * Get the registrations for a receiver; this comes with:
+   *  minfo: the method info the receiver stored
+   *  cinfo: the class info the receiver stored
+   *  method: the method registration (w/ workflow, transaction, function, and other info)
+   */
+  getRegistrationsFor(eri: DBOSEventReceiver) : {minfo: unknown, cinfo: unknown, method: MethodRegistrationBase}[];
+
+  transaction<T extends unknown[], R>(txn: Transaction<T, R>, params: WorkflowParams, ...args: T): Promise<R>;
+  workflow<T extends unknown[], R>(wf: WorkflowFunction<T, R>, params: WorkflowParams, ...args: T): Promise<WorkflowHandle<R>>;
+}
+
+/*
+ * Interface for receiving events
+ *  This is for things like kafka, SQS, etc., that poll for events and dispatch workflows
+ * Needs to be:
+ *  Registered with DBOS executor if any decorated endpoints need it
+ *  Initialized / destroyed with the executor
+ * It is the implememnter's job to keep going and dispatch workflows between those times
+ */
+export interface DBOSEventReceiver
+{
```

Comment:
Maybe put `executor` as a required field here? Then it'll remind people of defining an executor field when they implement the event receiver interface.

Reply:
It's a likely field, but when you go to use it, you'll know if you don't have it.  It's also hard to make it a truly required field, because the constructor doesn't know the value yet.

Reply:
A likely field sounds good, and it'll also make naming consistent.
In kafka.ts it uses `dbosExec` but in eventreceiver.test.ts it uses `executor`.

Reply:
As predicted, putting `executor?: DBOSExecutorContext` in the interface had no effect whatsoever.  I will rename in kafka though.